content/blog/is-dns-migration-a-tricky-affair.md

```diff
@@ -0,0 +1,106 @@
+We actually used Terraform to add the hosted zone as well as other records, as this would ease up the maintenance of the records.
+
+## Verification Phase
+This phase included verification of the application and mail server. One way to verify this is by simply heading to the application domain and sending a mail to the colleague. This can be considered as a correct way to verify since we have taken care of the TTL, although, to make sure the DNS query result is being served by Route53 nameservers we have to make use of a linux utility called - 
+
+![dig](/images/blog/is-dns-migration-a-ticky-affair/dig.gif)
+```bash
+	$ dig -t NS my_cool_domain.com
+```
+
+The response should have the Route53 nameservers and NOT the GoDaddy DNS nameservers. This validates that all the future queries for our domain/subdomain will be served by Route53.
+
+## Rollback Strategy
+There can be scenarios where the migration may go wrong, and it may take time for debugging the issue and the fix may take time. In such cases, we can revert to the old DNS service with a simple change (assuming that the records are not cleaned up from the old service). We only need to replace the Route53 nameservers with the old nameservers in the Domain Registrar.
+
+# D-Day
+It was this day (and a few of my colleagues 😝) that compelled me to write this blog.  We proceeded as per our well-thought-out plan (_at-least what we believed it to be_), everything worked well...until the verification phase. We were not able to send/receive emails and the application was down. Furthermore, we verified using **dig** as mentioned in the verification phase, and it was returning the Route53 nameservers as expected. This GIF can express what I was like that day.
+
+
+![me crying](/images/blog/is-dns-migration-a-ticky-affair/bug.gif)
+
+
+Others in my team were trying to console me by showing tweets of failed DNS migrations and how common this is. Just kidding!! They were like
+![panicing](/images/blog/is-dns-migration-a-ticky-affair/panic.gif)
+
+Well, this is a bit exaggerated. 
+# Troubleshooting
+With the verification using dig, we were sure that migration did happen and DNS had migrated to Route53. This eliminated the chance of issue in nameserver configuration at the domain registrar end. 
+Next we moved to Route53 to check the domain records for the email server and application. It was at this point we noticed something unusual.
+
+When we copied the records from the zone file exported from GoDaddy DNS service, we copied the name of the records as it is. In a lot of DNS services "@" is used to represent the apex domain i.e. **my_cool_domain.com**. We assumed it to be the same for Route53 as well and applied the changes. This led to creation of subdomain @.my_cool_domain.com in Route53.
```

Comment:
Invalid domain 😆 

s/my_cool_domain.com/my-cool-domain.in/g

Reply:
Fixed in 15acecd